### PR TITLE
Add debug logging

### PR DESCRIPTION
**Pull request by 2opremio**

### CODE REVIEW

File: app/controls.go [hunk 1]
```diff
@@ -9,6 +8,7 @@ import (
 
 	"github.com/gorilla/mux"
 
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline app/controls.go]: Never ever use dot imports

File: app/pipes.go [hunk 2]
```diff
@@ -2,13 +2,13 @@ package app
...
 import (
 	"io"
-	"log"
 	"net/http"
 	"sync"
 	"time"
 
 	"github.com/gorilla/mux"
 
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline app/pipes.go]: Never ever use dot imports

File: app/server_helpers.go [hunk 3]
```diff
@@ -2,8 +2,9 @@ package app
 
 import (
 	"encoding/json"
-	"log"
 	"net/http"
+
+	. "github.com/weaveworks/scope/common"
```
Comment (peterbourgon) [inline app/server_helpers.go]: Never ever use dot imports

File: common/sanitize/sanitize.go [hunk 4]
```diff
@@ -2,10 +2,11 @@ package sanitize
 
 import (
 	"fmt"
-	"log"
 	"net"
 	"net/url"
 	"strings"
+
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline common/sanitize/sanitize.go]: Never ever use dot imports

File: probe/appclient/app_client.go [hunk 5]
```diff
@@ -12,6 +11,7 @@ import (
 
 	"github.com/gorilla/websocket"
 
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline probe/appclient/app_client.go]: Never ever use dot imports

File: probe/appclient/multi_client.go [hunk 6]
```diff
@@ -6,10 +6,10 @@ import (
 	"fmt"
 	"io"
 	"io/ioutil"
-	"log"
 	"strings"
 	"sync"
 
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline probe/appclient/multi_client.go]: Never ever use dot imports

File: probe/appclient/resolver.go [hunk 7]
```diff
@@ -1,12 +1,12 @@
 package appclient
 
 import (
-	"log"
 	"net"
 	"strconv"
 	"strings"
 	"time"
 
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline probe/appclient/resolver.go]: Never ever use dot imports

File: probe/docker/container.go [hunk 8]
```diff
@@ -17,6 +16,7 @@ import (
 
 	docker "github.com/fsouza/go-dockerclient"
 
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline probe/docker/container.go]: Never ever use dot imports

File: probe/docker/container_test.go [hunk 9]
```diff
@@ -13,6 +12,7 @@ import (
 
 	client "github.com/fsouza/go-dockerclient"
 
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline probe/docker/container_test.go]: Never ever use dot imports

File: probe/docker/controls.go [hunk 10]
```diff
@@ -1,10 +1,9 @@
 package docker
 
 import (
-	"log"
-
 	docker_client "github.com/fsouza/go-dockerclient"
 
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline probe/docker/controls.go]: Never ever use dot imports

File: probe/docker/registry.go [hunk 11]
```diff
@@ -1,12 +1,12 @@
 package docker
 
 import (
-	"log"
 	"sync"
 	"time"
 
 	docker_client "github.com/fsouza/go-dockerclient"
 
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline probe/docker/registry.go]: Never ever use dot imports

File: probe/docker/reporter.go [hunk 12]
```diff
@@ -1,11 +1,11 @@
 package docker
 
 import (
-	"log"
 	"net"
 
 	docker_client "github.com/fsouza/go-dockerclient"
 
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline probe/docker/reporter.go]: Never ever use dot imports

File: probe/endpoint/conntrack.go [hunk 13]
```diff
@@ -4,13 +4,13 @@ import (
...
 	"encoding/xml"
 	"io"
-	"log"
 	"os"
 	"strings"
 	"sync"
 	"time"
 
 	"github.com/weaveworks/scope/common/exec"
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline probe/endpoint/conntrack.go]: Never ever use dot imports

File: probe/endpoint/procspy/proc_linux.go [hunk 14]
```diff
@@ -13,6 +12,7 @@ import (
 	"github.com/hashicorp/go-version"
 
 	"github.com/weaveworks/scope/common/fs"
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline probe/endpoint/procspy/proc_linux.go]: Never ever use dot imports

File: probe/kubernetes/client.go [hunk 15]
```diff
@@ -10,6 +9,8 @@ import (
 	"k8s.io/kubernetes/pkg/fields"
 	"k8s.io/kubernetes/pkg/labels"
 	"k8s.io/kubernetes/pkg/util"
+
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline probe/kubernetes/client.go]: Never ever use dot imports

File: probe/overlay/weave.go [hunk 16]
```diff
@@ -12,6 +11,7 @@ import (
 	"time"
 
 	"github.com/weaveworks/scope/common/exec"
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline probe/overlay/weave.go]: Never ever use dot imports

File: probe/probe.go [hunk 17]
```diff
@@ -1,12 +1,12 @@
 package probe
 
 import (
-	"log"
 	"sync"
 	"time"
 
 	"github.com/armon/go-metrics"
 
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline probe/probe.go]: Never ever use dot imports

File: prog/app.go [hunk 18]
```diff
@@ -15,6 +14,7 @@ import (
 	"github.com/weaveworks/weave/common"
 
 	"github.com/weaveworks/scope/app"
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline prog/app.go]: Never ever use dot imports

File: prog/probe.go [hunk 19]
```diff
@@ -18,6 +17,7 @@ import (
 	"github.com/weaveworks/weave/common"
 
 	"github.com/weaveworks/scope/common/hostname"
+	. "github.com/weaveworks/scope/common/log"
```
Comment (peterbourgon) [inline prog/probe.go]: Never ever use dot imports

**Comment by peterbourgon** [comment]
[re: dot imports](https://github.com/golang/go/wiki/CodeReviewComments#import-dot)

**Comment by 2opremio** [comment]
@peterbourgon Weave uses dots for logging throughout it's sourcecode (that's where I took this from). Anyways, I discussed it with @tomwilkie and I am going to use logrus directly, reusing weave's formatter for uniformity.

File: prog/app.go [hunk 20]
```diff
@@ -32,16 +32,19 @@ func appMain() {
 	var (
 		window    = flag.Duration("window", 15*time.Second, "window")
 		listen    = flag.String("http.address", ":"+strconv.Itoa(xfer.AppPort), "webserver listen address")
+		logLevel  = flag.String("log-level", "info", "logging threshold level: debug|info|warn|error|fatal|panic")
```
Comment (peterbourgon) [inline prog/app.go]: log.level, for consistency

File: prog/probe.go [hunk 21]
```diff
@@ -77,24 +77,27 @@ func probeMain() {
 		useConntrack       = flag.Bool("conntrack", true, "also use conntrack to track connections")
 		insecure           = flag.Bool("insecure", false, "(SSL) explicitly allow \"insecure\" SSL connections and transfers")
 		logPrefix          = flag.String("log.prefix", "<probe>", "prefix for each log line")
+		logLevel           = flag.String("log-level", "info", "logging threshold level: debug|info|warn|error|fatal|panic")
```
Comment (peterbourgon) [inline prog/probe.go]: Likewise log.level

File: prog/probe.go [hunk 22]
```diff
@@ -77,24 +77,27 @@ func probeMain() {
...
 	)
 	flag.Parse()
 
+	setLogLevel(*logLevel)
+
+	if !strings.HasSuffix(*logPrefix, " ") {
+		*logPrefix += " "
+	}
```
Comment (peterbourgon) [inline prog/probe.go]: This can be done in setLogFormatter, I imagine?

File: prog/app.go [hunk 23]
```diff
@@ -32,16 +32,19 @@ func appMain() {
...
 		logPrefix = flag.String("log.prefix", "<app>", "prefix for each log line")
 	)
 	flag.Parse()
 
+	setLogLevel(*logLevel)
+
 	if !strings.HasSuffix(*logPrefix, " ") {
 		*logPrefix += " "
 	}
```
Comment (peterbourgon) [inline prog/app.go]: Likewise, can move to setLogFormatter?

**Comment by peterbourgon** [comment]
I'd vote for not aliasing the logrus import, but if you feel strongly that it should be aliased to `log` I guess that's fine. With those things fixed, LGTM.

**Pull request merged**